File: Updog.Application/Post/UseCases/Create/PostCreateParams.cs
using Updog.Domain;

namespace Updog.Application {
    /// <summary>
    /// Parameters to add a new post.
    /// </summary>
    public sealed class PostCreateParams {
        #region Properties
        public PostType Type { get; }

        public string Title { get; }

        public string Body { get; }

        public string SpaceName { get; }

        public User User { get; }

        #endregion

        #region Constructor(s)
        public PostCreateParams(PostType type, string title, string body, string spaceName, User user) {
            Type = type;
            Title = title;
            Body = body;
            SpaceName = spaceName;
            User = user;
        }
        #endregion
    }
}

File: Updog.Client/src/post/interactors/create/post-create-params.ts
import { PostType } from '@/post/domain/post-type';

/**
 * Request to create a new post.
 */
export class PostCreateParams {
    /**
     *
     * @param space The space to submit to.
     * @param type The type of post it is.
     * @param title The title of the post.
     * @param body The body of the post.
     */
    constructor(public space: string, public type: PostType, public title: string, public body: string) {}
}
